azurerm/resource_arm_analysis_services_server.go

```diff
@@ -0,0 +1,449 @@
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Azure ARM Analysis Services Server creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		server, err := client.GetDetails(ctx, resGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(server.Response) {
+				return fmt.Errorf("Error checking for presence of existing Analysis Services Server %q (Resource Group %q): %s", name, resGroup, err)
+			}
+		}
+
+		if server.ID != nil && *server.ID != "" {
+			return tf.ImportAsExistsError("azurerm_analysis_services_server", *server.ID)
+		}
+	}
+
+	sku := d.Get("sku").(string)
+	location := azure.NormalizeLocation(d.Get("location").(string))
+
+	serverProperties := expandAnalysisServicesServerProperties(d)
+
+	if fwSettings := serverProperties.IPV4FirewallSettings; len(*fwSettings.FirewallRules) > 0 && fwSettings.EnablePowerBIService == nil {
+		return fmt.Errorf("`enable_power_bi_service` must be set if there is at least one firewall rule")
+	}
+
+	tags := d.Get("tags").(map[string]interface{})
+
+	analysisServicesServer := analysisservices.Server{
+		Name:             &name,
+		Location:         &location,
+		Sku:              expandAnalysisServicesServerSku(sku),
+		ServerProperties: serverProperties,
+		Tags:             expandTags(tags),
+	}
+
+	future, err := client.Create(ctx, resGroup, name, analysisServicesServer)
+	if err != nil {
+		return fmt.Errorf("Error creating Analysis Services Server %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for completion of Analysis Services Server %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	server, err := client.GetDetails(ctx, resGroup, name)
+	if err != nil {
+		return err
+	}
+
+	if server.ID == nil {
+		return fmt.Errorf("Cannot read ID of Analysis Services Server %q (Resource Group %q)", name, resGroup)
+	}
+
+	d.SetId(*server.ID)
+
+	return resourceArmAnalysisServicesServerRead(d, meta)
+}
+
+func resourceArmAnalysisServicesServerRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).analysisServicesServerClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resGroup := id.ResourceGroup
+	name := id.Path["servers"]
+
+	server, err := client.GetDetails(ctx, resGroup, name)
+
+	if err != nil {
+		if utils.ResponseWasNotFound(server.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Azure Analysis Services Server %q: %+v", name, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resGroup)
+
+	if location := server.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	d.Set("sku", flattenAnalysisServicesServerSku(server.Sku))
```

Comment:
This should simply be:
```suggestion
	d.Set("sku", server.Sku)
```

Reply:
done